keps/sig-node/4800-cpumanager-split-uncorecache/README.md

```diff
@@ -0,0 +1,689 @@
+
+#### Story 2
+
+- As a Networking/Telco Engineer, my application is sensitive to uncore cache latency. I want my application to be statically assigned to CPUs that correspond to one single uncore cache so that I can get the maximum throughput from my application. Additionally, I have applications that are also static and of different CPU size requirements. The default Static CPU Manager Policy will result in my applications being assigned to two or more uncore caches. I have a large multi-node Kuberentes cluster. I want a scheduling policy to automatically assign CPUs for the best performance across the fewest amount of uncore cahces based on available CPUs without having to worry about what order/size I deploy my applications so I can ensure I get the best performance possible. 
+
+### Notes/Constraints/Caveats (Optional)
+
+- The name UnCoreCache is directly derived from cAdvisor that is being used as a package. 
+- Using the flow in `CPUManagerPolicy{Alpha,Beta}Options` from [xref](https://github.com/kubernetes/kubernetes/blob/master/pkg/features/kube_features.go#L110C1-L117C83) which is used to avoid a proliferation of a feature gates. 
+
+### Risks and Mitigations
+
+- Feature is enabled by a static policy option flag.  
+- It does not change behavior of non-static policy
+- It preserves the behavior of other static options.
+- Failure will occur if mismatch between options occurs and the Pod will not be scheduled.
+
+## Design Details
+
+To make Kubelet be uncore cache aware for CPU sensitive workloads the CPU allocation, `UnCoreCacheID` has been added to the `CPUInfo` and `CPUTopology` structures that are provided by `cAdvisor` during the CPU discovery process.
+
+A new CPUManager policy option, align-cpus-by-uncorecache, will be introduced to enable this feature. When enabled, the existing default scheduling function will be modified to account for UnCoreCache as a preference. The allocation process will first attempt to allocate CPUs from an entire socket. Next, it will attempt to allocate CPUs within the logical boundary of a NUMA node within the socket. Finally, the allocation will consider the subset of CPUs associated with each UnCoreCache within the previously selected CPUs. While UnCoreCache allocation is preferred, it is not strictly required this policy will not fail if alignment is not possible. Since this policy extends the CPUManager’s Static policy, it will only apply to guaranteed pods with whole CPU requests.
```

Comment:
same comment about possibly replacing `scheduling` with `cpu allocation logic`. I can't remember us using "scheduling" in this context in the kubelet lingo.

Reply:
updated